builder/builder.go

```diff
@@ -498,7 +502,8 @@ func (b *Builder) convertAndBuild(ctx context.Context, target domain.Target, opt
 			console := b.opt.Console.WithPrefixAndSalt(sts.Target.String(), sts.ID)
 			for _, saveImage := range sts.SaveImages {
 				doSave := (sts.GetDoSaves() || saveImage.ForceSave)
-				shouldPush := opt.Push && saveImage.Push && !sts.Target.IsRemote() && saveImage.DockerTag != "" && doSave
+				doPush := (sts.GetDoPushes() || saveImage.ForceSave)
```

Comment:
What is `ForceSave` used for?

Reply:
Good point, I think I might be able to drop it.

It looks like it's just for older version:
```
	// ForceSaveImage is used to force all SAVE IMAGE commands are executed regardless of if they are
	// for a local or remote target; this is to support the legacy behaviour that was first introduced in earthly (up to 0.5)
	// When this is set to false, SAVE IMAGE commands are only executed when DoSaves is true.
```

```
# No Version
pi:
    FROM ubuntu:latest
    RUN apt-get update && apt-get install -y bc
    ARG --required iterations
    RUN seq -f '4/%g' 1 2 $iterations | paste -sd-+ | bc -l > /root/pi
    CMD cat /root/pi
    SAVE ARTIFACT /root/pi
    SAVE IMAGE --push alexcb132/pi:$iterations

indirect:
    FROM +pi
    RUN cat /root/pi
```

running this via: `/home/alex/gh/earthly/earthly/build/linux/amd64/earthly --push +indirect --iterations=11` still results in a push even once I have removed `ForceSave`.

I'll do a few more tests before submitting an update with this removed.

Reply:
I double checked that

```
test:
    BUILD github.com/alexcb/earthly-push-tests:poorly-hidden-branch+pi --iterations=100
```

doesn't propagate a push, so we should be safe to remove `|| ForceSave` from the condition.